packages/services/src/models/Etherscan.js

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios'
+import { bn } from '@aragonone/court-backend-shared/helpers/numbers'
+
+const BASE_URL = 'https://<network>.etherscan.io/api?'
+
+export default class Etherscan {
+  constructor(network, key) {
+    this.key = key
+    this.url = this._getBaseUrl(network)
+  }
+
+  async getBalance(address) {
+    const path = `${this.url}module=account&action=balance&address=${address}&tag=latest`
+    const response = await this._get(path)
+    if (response.status !== '1' || !response.result) throw Error(`Received error from Etherscan: ${JSON.stringify(response)}`)
+    return bn(response.result)
+  }
+
+  async getTransactionsFrom(address, fromBlock) {
+    const path = `${this.url}module=account&action=txlist&address=${address}&startblock=${fromBlock}&sort=desc`
+    const response = await this._get(path)
+
+    if ((response.status !== '1' && response.message !== 'No transactions found') || !response.result)
```

Comment:
This status code check should probably be for `'200'`? Or does Etherscan return `1` as an HTTP status code?

Reply:
Note that this `status` is part of the response data, not the HTTP header. Try these queries:
1. [successful](http://api.etherscan.io/api?module=account&action=txlist&address=0xde0b295669a9fd93d5f28d9ec85e40f4cb697bae&startblock=0&endblock=99999999&sort=asc&apikey=YourApiKeyToken): you will see that `status` is `1` here
2. [failure](http://api.etherscan.io/api?module=account&action=txlist&address=0xfe0b295669a9fd93d5f28d9ec85e40f4cb697bae&startblock=0&endblock=99999999&sort=asc&apikey=YourApiKeyToken): you will see that `status` is `0` here

They also provide an `isError` field, but as you can see this does variable does not appear in the second scenario.